src/Server/InterserverIOHTTPHandler.cpp

```diff
@@ -117,7 +117,17 @@ void InterserverIOHTTPHandler::handleRequest(HTTPServerRequest & request, HTTPSe
         if (auto [message, success] = checkAuthentication(request); success)
         {
             processQuery(request, response, used_output);
-            used_output.out->finalize();
+
+            try
+            {
+                used_output.out->finalize();
```

Comment:
I suppose we also have to destroy buffer in catch session, otherwise in the debug build the check "WriteBuffer is not finalized when destructor is called." would be triggered.

Reply:
No. Im wrong, `finalized = true;` is set for even failed attempt to call `finalize()`.

Reply:
`finalized` is set to true even if exception happened so it should be okay.

Reply:
I think that we could encounter similar cases in lines. The case when something wring on server side but after we unable to send error to the client as well.

https://github.com/ClickHouse/ClickHouse/blob/f421d3d571e07c2d4a5d006128fe2d2b54b82b35/src/Server/InterserverIOHTTPHandler.cpp#L99

https://github.com/ClickHouse/ClickHouse/blob/f421d3d571e07c2d4a5d006128fe2d2b54b82b35/src/Server/InterserverIOHTTPHandler.cpp#L111

Reply:
yes you are correct